File: DAY ONE/PYTHON/arrays_check.py
import math

def even_odd(lst):
    even = []
    odd = []

    for num in lst:
        if num % 2 == 0:
            even.append(num)
        else:
            odd.append(num)

    return [odd, even]


def is_palindrome(lst):
    left = 0
    right = len(lst) - 1

    while left < right:
        if lst[left] != lst[right]:
            return False

        left += 1
        right -= 1

    return True


def perfect_square(lst):
    result = []

    for num in lst:
        square_root = int(math.sqrt(num))

        if square_root * square_root == num:
            result.append(num)

    return result


def non_perfect_square(lst):
    for index in range(len(lst)):
        square_root = int(math.sqrt(lst[index]))

        if square_root * square_root != lst[index]:
            lst[index] = -1

    return lst

File: DAY ONE/JAVASCRIPT/arrayscheck.js
function evenOdd(list) {
    let even = [];
    let odd = [];

    for (let index = 0; index < list.length; index++) {
        if (list[index] % 2 === 0) {
            even.push(list[index]);
        } else {
            odd.push(list[index]);
        }
    }

    return [odd, even];
}


function isPalindrome(list) {
    let left = 0;
    let right = list.length - 1;

    while (left < right) {
        if (list[left] !== list[right]) {
            return false;
        }

        left++;
        right--;
    }

    return true;
}


function perfectSquare(list) {
    let result = [];

    for (let index = 0; index < list.length; index++) {
        let squareRoot = Math.floor(Math.sqrt(list[index]));

        if (squareRoot * squareRoot === list[index]) {
            result.push(list[index]);
        }
    }

    return result;
}


function nonPerfectSquare(list) {
    for (let index = 0; index < list.length; index++) {
        let squareRoot = Math.floor(Math.sqrt(list[index]));

        if (squareRoot * squareRoot !== list[index]) {
            list[index] = -1;
        }
    }

    return list;
}
